### ISSUE TITLE
Validator produces repeated property names when validating an object inside an array

### ISSUE THREAD

**Issue by eduramiba** [raml-org/raml-js-parser-2#414]
For example, if the ```kind``` property is invalid for API https://github.com/raml-org/raml-examples/blob/master/typesystem/complex.raml in ```Head.reports[0]```, the error ```vp``` field will contain:

```json
{"name":"Head","child":{"name":"reports","child":{"name":"0","child":{"name":"kind","child":{"name":"kind"}}}}}
```

Note kind appears two times.

Version used 0.2.31

Thank you

**Comment by eduramiba** [comment]
Is this the expected behavior?

Thanks

**Comment by ddenisenko** [comment]
Not sure how to reproduce this. Could you please provide the complete RAML with an error and the complete output?

**Comment by eduramiba** [comment]
Sure, I upload a complete example, reproducible with latest version (1.1.2)
[example.zip](https://github.com/raml-org/raml-js-parser-2/files/474084/example.zip)

Thanks

**Comment by ddenisenko** [comment]
Thank you for the example.

So, what I see in the parser output in this case of `phone` property indentation being incorrect:

1) Two errors: one for the missing `phone` property in the example, another one for the unknown node `phone`. Looks fine:

```
  "errors": [
    {
      "code": 11,
      "message": "Required property: phone is missed",
      "path": "complex.raml",
      "range": {
        "start": {
          "line": 49,
          "column": 14,
          "position": 1611
        },
        "end": {
          "line": 49,
          "column": 18,
          "position": 1615
        }
      }
    },
    {
      "code": 2,
      "message": "Unknown node: phone",
      "path": "complex.raml",
      "range": {
        "start": {
          "line": 58,
          "column": 0,
          "position": 1856
        },
        "end": {
          "line": 58,
          "column": 5,
          "position": 1861
        }
      }
    }
  ]
```

2) Parsed example containing adequate JSON representation of the example:

```
                      "structuredValue": {
                        "onCall": {
                          "firstname": "nico",
                          "lastname": "ark",
                          "kind": "admin",
                          "clearanceLevel": "low",
                          "phone": "12321"
                        },
                        "Head": {
                          "firstname": "nico",
                          "lastname": "ark",
                          "kind": "manager",
                          "reports": [
                            {
                              "firstname": "nico",
                              "lastname": "ark",
                              "kind": "admin"
                            }
                          ]
                        }
                      }
```

Looks also okay for me.

Attached is  the complete print of api.toJSON() for reference.
[toJSONPrint.txt](https://github.com/raml-org/raml-js-parser-2/files/475945/toJSONPrint.txt)

Please let me know, how you obtain the data, which sample was printed in the initial post.
I mean, the JS code, which obtains the data from the parser output.

My guess is that you are doing something different than calling `api.toJSON()` and checking the errors in the output or setting `rejectOnErrors: true` in the parsing options and then analyzing the thrown error object.

**Comment by eduramiba** [comment]
Hi, sorry, I think I uploaded the RAML with an indentation error by accident. I upload it again. [example.zip](https://github.com/raml-org/raml-js-parser-2/files/476351/example.zip)

I am using the api to programmatically validate custom data (and I am not using api.toJSON at all).
For doing that, I am using the api type node's `validateInstanceWithDetailedStatuses` method.

In this case the type node is the response body node.
Then, my data being validated is:

```javascript
{
    onCall: {
        firstname: 'nico',
        lastname: 'ark',
        kind: 'admin',
        clearanceLevel: 'low',
        phone: '12321',
        title: 'Something 1'
    },
    Head: {
        firstname: 'nico',
        lastname: 'ark',
        kind: 'manager',
        title: 'Something 2',
        reports: [
            {
                firstname: 'nico',
                lastname: 'ark',
                kind: 12,//This is invalid
                title: 'Something 3'
            }
        ],
        phone: '123-23'
    }
}
```

Where I deliberately make `kind` have the wrong type.
And I would expect that method to return in its `vp` property:

```json
{
  "name": "Head",
  "child": {
    "name": "reports",
    "child": {
      "name": "0",
      "child": {
        "name": "kind"
      }
    }
  }
}
```

but it returns:

```json
{
  "name": "Head",
  "child": {
    "name": "reports",
    "child": {
      "name": "0",
      "child": {
        "name": "kind",
        "child": {
          "name": "kind"
        }
      }
    }
  }
}
```

**Comment by ddenisenko** [comment]
Okay, got it.

Yes, I think validation path should be "Head/reports/0/kind" instead of "Head/reports/0/kind/kind" in this case. 
Classified as bug.

Btw, you can can the path string by calling `getValidationPathAsString` method of the status.

**Comment by eduramiba** [comment]
This seems to be fixed with version 1.1.4

**Comment by eduramiba** [comment]
Maybe it was fixed indirectly, also works fine in 1.1.5 and 1.1.6.
